log4j-core/src/main/java/org/apache/logging/log4j/core/async/AsyncLoggerConfigDisruptor.java

```diff
@@ -155,6 +165,16 @@ private void notifyIntermediateProgress(final long sequence) {
                 ringBufferElement.loggerConfig = loggerConfig;
             };
 
+    private Log4jEventWrapperHandler createEventHandler() {
+        try {
+            return LoaderUtil.newInstanceOf(
+                    "org.apache.logging.log4j.core.async.AsyncLoggerConfigDisruptor$Log4jEventWrapperHandler3");
+        } catch (final ReflectiveOperationException | LinkageError e) {
+            LOGGER.debug("LMAX Disruptor 3.x is missing, trying version 4.x.", e);
+        }
+        return new Log4jEventWrapperHandler();
```

Comment:
1. Doesn't this imply that Disruptor 4 users will always get a debug message telling the obvious?
2. Can't we rather `switch` on `DisruptorUtil.DISRUPTOR_MAJOR_VERSION` and _only_ support 3 and 4?